test/suite/run.py

```diff
@@ -133,6 +132,32 @@ def usage():
   When -s is present, there must be a test named.\n\
 ')
 
+# Find an executable of the given name in the execution path.
+def which(name):
+    path = os.getenv('PATH')
+    for pathdir in path.split(os.path.pathsep):
+        fname = os.path.join(pathdir, name)
+        if os.path.exists(fname) and os.access(fname, os.X_OK):
+            return fname
+    return None
+
+# Follow a symbolic link, returning the target
+def follow_symlinks(pathname):
+    return os.path.realpath(pathname)
+
+# Find all instances of a filename under a directory
+def find(topdir, filename):
```

Comment:
Sorry, I'm not familiar with os.walk. Now I understand what you are trying to do.